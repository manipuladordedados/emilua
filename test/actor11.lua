local sleep_for = require('sleep_for')

if _CONTEXT == 'main' then
    local ch = spawn_vm('.')
    local f = spawn(function()
        ch:send('foobar')
    end)
    sleep_for(100)
    f:interrupt()
    local ok, e = pcall(function() f:join() end)
    print(f.interruption_caught)
    error(e)
else assert(_CONTEXT == 'worker')
    spawn(function()
        sleep_for(200)
    end):detach()
end
