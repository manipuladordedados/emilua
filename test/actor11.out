false
Main fiber from VM 0x1 panicked: 'Channel closed'
stack traceback:
	input.lua:12: in main chunk
	[C]: in function ''
	[string "?"]: in function <[string "?"]:0>
